src/hypothesis/strategies.py

```diff
@@ -1038,11 +1039,20 @@ def from_type(thing):
             # Under Python 3.6, Unions are not instances of `type` - but we
             # still want to resolve them!
             if getattr(thing, '__origin__', None) is typing.Union:
-                args = sorted(thing.__args__, key=types.type_sorting_key)
-                return one_of([from_type(t) for t in args])
+                args = {t: types.deferred_resolve(t) for t in thing.__args__}
+                return one_of([p[1] for p in sorted(
+                    args.items(),
+                    key=lambda p: types.type_sorting_key(p[0])
+                )])
         except ImportError:  # pragma: no cover
             pass
         raise InvalidArgument('thing=%s must be a type' % (thing,))
+    elif sys.version_info[:2] == (3, 5):  # pragma: no cover
```

Comment:
Is this code actually tested? I know it won't show up in coverage either way and that's fine of course, but it would be easy to write a test.

Reply:
Also is this version check actually needed? The code would be simpler and more future proof without it if they reverse their decision and making `_ForwardRef` a type again.